llvm/test/CodeGen/AMDGPU/promote-alloca-vector-to-vector.ll

```diff
@@ -170,7 +173,9 @@ entry:
 ; OPT-LABEL: define i64 @ptr_alloca_bitcast
 
 ; GCN-NOT: buffer_
-; GCN: v_mov_b32_e32 v1, 0
+; GCN: s_waitcnt vmcnt(0) expcnt(0) lgkmcnt(0)
+; GCN-NEXT: s_setpc_b64
+
```

Comment:
Because the test is not auto generated it's not really obvious what is happening here.
I assume v1 is not uninitialized at function return?

Reply:
Yes, this now optimizes out fully and avoids the previous avoidable zero initialize